platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,194 @@
+
+    // set up metadata
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+    Iterator<Map.Entry<String, EntityInformation>> it = testMap.entrySet().iterator();
+
+    // get the second metadata
+    Map.Entry<String, EntityInformation> testEntry = it.next();
+    entityID2 = testEntry.getKey();
+    testInfo2 = testEntry.getValue();
+
+    // get the first metadata
+    testEntry = it.next();
+    entityID = testEntry.getKey();
+    testInfo = testEntry.getValue();
+  }
+
+  @Test
+  public void parseNullMetadata() throws Exception {
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(null, bindingSet);
+
+    assertThat(testMap, is(equalTo(EMPTY_MAP)));
+  }
+
+  @Test
+  public void testEntityIDs() throws Exception {
+    assertThat(entityID, is(equalTo(ENTITY_ID)));
+    assertThat(entityID2, is(equalTo(ENTITY_ID_2)));
+  }
+
+  @Test
+  public void testSigningCertificates() {
+    assertThat(testInfo.getSigningCertificate(), is(equalTo(CERTIFICATE)));
+    assertThat(testInfo2.getSigningCertificate(), is(equalTo(CERTIFICATE)));
+  }
+
+  @Test
+  public void testEncryptionCertficates() {
+    assertThat(testInfo.getEncryptionCertificate(), is(equalTo(CERTIFICATE)));
+    assertThat(testInfo2.getEncryptionCertificate(), is(equalTo(CERTIFICATE)));
+  }
+
+  @Test
+  public void testRedirectLogoutUrls() {
+    assertThat(
+        testInfo.getLogoutService(Binding.HTTP_REDIRECT).getUrl(),
+        is(equalTo(REDIRECT_LOGOUT_URL)));
+    assertThat(
+        testInfo2.getLogoutService(Binding.HTTP_REDIRECT).getUrl(),
+        is(equalTo(REDIRECT_LOGOUT_URL)));
+  }
+
+  @Test
+  public void testPostLogoutUrls() {
+    assertThat(testInfo.getLogoutService(Binding.HTTP_POST).getUrl(), is(equalTo(POST_LOGOUT_URL)));
+    assertThat(
+        testInfo2.getLogoutService(Binding.HTTP_POST).getUrl(), is(equalTo(POST_LOGOUT_URL)));
+  }
+
+  @Test
+  public void testRedirectAssertionUrls() {
+    assertThat(
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, null).getUrl(),
+        is(equalTo(REDIRECT_ACS_URL)));
+    assertThat(
+        testInfo2.getAssertionConsumerService(null, Binding.HTTP_REDIRECT, null).getUrl(),
+        is(equalTo(REDIRECT_ACS_URL)));
+  }
+
+  @Test
+  public void testPostAssertionUrls() {
+    assertThat(
+        testInfo.getAssertionConsumerService(null, Binding.HTTP_POST, null).getUrl(),
+        is(equalTo(POST_ACS_URL)));
+    assertThat(
+        testInfo2.getAssertionConsumerService(null, Binding.HTTP_POST, null).getUrl(),
+        is(equalTo(POST_ACS_URL)));
+  }
+
+  @Test
+  public void testSoapAssertionUrl() {
+    assertThat(
+        testInfo.getAssertionConsumerService(null, Binding.SOAP, null).getUrl(),
+        is(equalTo(SOAP_ACS_URL)));
+  }
+
+  @Test
+  public void testNoSoapAssertionUrl() {
+    assertThat(
+        "Expected to not find a SOAP binding, and for the method to return the default (Post) binding instead",
```

Comment:
✅  Excellent use of the failure message!!!
❓ Could we concatenate the values under test as part of this string?